components/js/SlickGrid.ts

```diff
@@ -144,16 +132,60 @@ export class SlickGrid implements OnChanges, OnInit, OnDestroy, AfterViewInit {
 
     private _grid: any;
     private _gridColumns: ISlickGridColumn[];
+    private _columnNameToIndex: any;
     private _gridData: ISlickGridData;
     private _rowHeight = 29;
     private _resizeSubscription: Subscription;
     private _gridSyncSubscription: Subscription;
     private _topRow: number = 0;
     private _leftPx: number = 0;
+    private _activeEditingRow: number;
+    private _activeEditingRowHasChanges: boolean;
     /* andresse: commented out 11/1/2016 due to minification issues
     private _finishGridEditingFn: (e: any, args: any) => void;
     */
 
+    @Output() public enterEditSession(): void {
+        this.enableEditing = true;
```

Comment:
Minor: since this and endEditSession have almost the same logic, have a private function they both call.